### PR TITLE
Add JS charts page

### PR DESCRIPTION
**Pull request by Ksinia**
# AnyChart Charts Page
- The left chart shows all the files and the top 10 files downloaded within a certain period until the current moment.
- The right chart shows all nodes and the top 10 nodes from which the files were downloaded within a certain period until the current moment.
- The frequency and format of the x-axis labels vary depending on the length of the period.
### Screenshots:
- The default view (10 days period):
Screenshot 2020-02-20 at 23 17 45

- 1 hour selected, tooltip is shown on hover:
Screenshot 2020-02-20 at 23 26 49

- View on narrow screens, 12 hours selected:
Screenshot 2020-02-20 at 23 19 30

### CODE REVIEW

**Comment by paskal** [comment]
I'll need to create separate PR to utilize this page after the merge.

File: webapp/index.js [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+
+function transformToFiles(jsonData) {
+  const filesData = jsonData.reduce(
+    (acc, element) => {
+      acc.files[0].data.push([
+        new Date(element.StartMinute).getTime(),
+        element.Nodes.all.Volume
+      ]);
+      Object.keys(element.Nodes.all.Files).forEach(file => {
+        if (!acc.files.find(el => el.file === file)) {
```
Comment (Mavrin) [inline webapp/index.js]: This is looks like O(N^3). May be it good for this case or not.

File: webapp/index.js [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+    )}`
+  );
+  drawChart(
+    data.nodes,
+    "top-nodes",
+    `Top 10 nodes from ${from.toLocaleDateString([], dateTimeOptions)}`
+  );
+}
+
+anychart.onDocumentLoad(() => {
```
Comment (Mavrin) [inline webapp/index.js]: Is not need document already loaded.

File: webapp/index.js [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+  loadAndDraw();
+});
+
+const buttons = document.getElementsByClassName("period-button");
```
Comment (Mavrin) [inline webapp/index.js]: You can use https://javascript.info/event-delegation or https://learn.javascript.ru/event-delegation and use https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes

File: webapp/index.js [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+for (let button of buttons) {
+  button.onclick = event => {
+    let args = {};
+    if (event.target.id === "1hour") {
```
Comment (akellbl4) [inline webapp/index.js]: I prefer `switch...case` for situations like this.

File: webapp/index.js [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+  chart.legend(true);
+  // set the layout of the legend
+  chart.legend().itemsLayout("vertical-expandable");
+  // set the position of the legend
+  chart.legend().position("bottom");
+  // set the maximum height and width of the legend
+  chart.legend().height("220px");
+  chart.height("500px");
+  const margin = chart.legend().margin();
+  margin.top(10);
```
Comment (akellbl4) [inline webapp/index.js]: `margin` used only one time I think it is not necessary to define var for it.

Looks better to me:
```js
chart.legend().margin().top(10)
```

File: webapp/index.js [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+  const sortedNodesData = filesData.nodes.sort((a, b) => {
+    return (
+      b.data.reduce((acc, element) => {
+        return (acc += element[1]);
+      }, 0) -
+      a.data.reduce((acc, element) => {
+        return (acc += element[1]);
+      }, 0)
+    );
+  });
```
Comment (akellbl4) [inline webapp/index.js]: Sort functions look the same.
It can be moved to a function.

File: webapp/index.js (hunk 2 above)
Comment (Ksinia) [reply webapp/index.js]: Is the current version better?

Comment (Mavrin) [reply webapp/index.js]: Yes

**Comment by paskal** [comment]
AnyChart license is no good: user need to require permissions to use it for non-commercial purposes. We need to pick something else.

**Comment by paskal** [comment]
@Mavrin would you be so kind as to check what's going on with Tauchart on resize?

To run:
```
#checkout this branch first
docker-compose build
docker-compose up -d

while true; do
  curl "http://127.0.0.1:7070/api/v1/jump/test1?url=/mp3files/test_file_$(((RANDOM % 10) + 1)).mp3" >/dev/null 2>&1
  sleep $(((RANDOM % 10) + 1))
done
```
Then head to http://localhost:8080/ and try to resize the page: everything works more or less fine (except for strange legend for tauchart).

But when you'll click any of the buttons (let's say, 1 hour) multiple times and then resize the screen even slightly, tauchart will re-render itself amount of times you clicked the button.

File: webapp/index.html [hunk 7]
```diff
@@ -0,0 +1,52 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <meta http-equiv="X-UA-Compatible" content="ie=edge" />
+    <title>Document</title>
+    <script
```
[the remark this answers is not in the document]
Comment (Ksinia) [reply webapp/index.html]: Added 3 more MIT license charts for comparison.

**Comment by Ksinia** [comment]
Dear @umputun, please try out current version of PR. I would prefer Apexcharts both visually and from a programming perspective.

**Pull request merged**